aarch64/kernel/memory/src/paging/table.rs

```diff
@@ -0,0 +1,177 @@
+            let table_address = self as *const _ as usize;
+            let next_table_vaddr: usize = match active {
+
+                // use the recursive entry
+                true => (table_address << 9) | (index << PAGE_SHIFT),
+
+                // use the identity mapping
+                false => self[index].pointed_frame().unwrap().start_address().value(),
+
+            };
+            Some(VirtualAddress::new_canonical(next_table_vaddr))
+        } else {
+            None
+        }
+    }
+
+    /// Returns a reference to the next lowest-level page table.
+    /// 
+    /// A convenience wrapper around `next_table_address()`; see that method for more.
+    /// 
+    /// With `active` set to true, you indicate that the
+    /// modified table is currently used by the CPU, in which case
+    /// it will be accessed using the recursive paging special entry.
+    /// Otherwise, the code assumes we are in identity-mapping, and
+    /// the physical addresses in a table are used as virtual addresses.
+    pub fn next_table(&self, index: usize, active: bool) -> Option<&Table<L::NextLevel>> {
+        // convert the next table address from a raw pointer back to a Table type
+        self.next_table_address(index, active).map(|vaddr| unsafe { &*(vaddr.value() as *const _) })
+    }
+
+    /// Returns a mutable reference to the next lowest-level page table.
+    /// 
+    /// A convenience wrapper around `next_table_address()`; see that method for more.
+    /// 
+    /// With `active` set to true, you indicate that the
+    /// modified table is currently used by the CPU, in which case
+    /// it will be accessed using the recursive paging special entry.
+    /// Otherwise, the code assumes we are in identity-mapping, and
+    /// the physical addresses in a table are used as virtual addresses.
+    pub fn next_table_mut(&mut self, index: usize, active: bool) -> Option<&mut Table<L::NextLevel>> {
+        self.next_table_address(index, active).map(|vaddr| unsafe { &mut *(vaddr.value() as *mut _) })
+    }
+
+    /// Returns a mutable reference to the next lowest-level page table, 
+    /// creating and initializing a new one if it doesn't already exist.
+    /// 
+    /// A convenience wrapper around `next_table_address()`; see that method for more.
+    /// 
+    /// With `active` set to true, you indicate that the
+    /// modified table is currently used by the CPU, in which case
+    /// it will be accessed using the recursive paging special entry.
+    /// Otherwise, the code assumes we are in identity-mapping, and
+    /// the physical addresses in a table are used as virtual addresses.
+    pub fn next_table_create(
+        &mut self,
+        index: usize,
+        flags: PteFlags,
+        active: bool,
+    ) -> Result<&mut Table<L::NextLevel>, &'static str> {
+        if self.next_table(index, active).is_none() {
+            // commenting until we understand how huge pages work on aarch64
```

Comment:
all the stuff about huge pages should be cfg-gated for x86 only in the latest changes from the main branch